marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -82,49 +90,229 @@ renderCurrentState walletDetails state =
+    mNickname = join $ Map.lookup party (state ^. _participants)
+  in
+    capitalize case party /\ mNickname of
+      -- TODO: For the demo we wont have PK, but eventually we probably want to limit the amount of characters
+      PK publicKey /\ _ -> publicKey
+      Role roleName /\ Just nickname -> roleName <> " (" <> nickname <> ")"
+      Role roleName /\ Nothing -> roleName
 
-    roles :: Array String
-    roles = Set.toUnfoldable $ Set.mapMaybe getRoleEntry (getParties contract)
+renderPartyTasks :: forall p. State -> Party -> Array NamedAction -> HTML p Action
+renderPartyTasks state party actions =
+  let
+    isActiveParticipant = (state ^. _mActiveUserParty) == Just party
 
-    -- FIXME: We fake the namedActions for development until we fix the semantics
-    actions' =
-      [ MakeDeposit (Role "into account") (Role "by") (Token "" "") $ fromInt 1500
-      ]
+    actionsSeparatedByOr =
+      intercalate
+        [ div [ classNames [ "font-semibold", "text-center", "my-2", "text-xs" ] ] [ text "OR" ]
+        ]
+        (Array.singleton <<< renderAction isActiveParticipant <$> actions)
+
+    participantName = participantWithNickname state party
+  in
+    div [ classNames [ "mt-3" ] ]
+      ( [ div [ classNames [ "text-xs", "flex", "mb-2" ] ]
+            -- TODO: In zeplin all participants have a different color. We need to decide how are we going to assing
+            --       colors to users. For now they all have blue
+            [ div [ classNames [ "bg-gradient-to-r", "from-blue", "to-lightblue", "text-white", "rounded-full", "w-4", "h-4", "text-center", "mr-1" ] ] [ text $ String.take 1 participantName ]
+            , div [ classNames [ "font-semibold" ] ] [ text participantName ]
+            ]
+        ]
+          <> actionsSeparatedByOr
+      )
+
+renderAction :: forall p. Boolean -> NamedAction -> HTML p Action
+renderAction isActiveParticipant (MakeDeposit intoAccountOf by token value) =
+  div_
+    [ shortDescription isActiveParticipant "chocolate pastry apple pie lemon drops apple pie halvah FIXME"
+    , button
+        [ classNames $ [ "flex", "justify-between", "px-6", "font-bold", "w-full", "py-5", "mt-2", "rounded-3xl", "shadow" ]
+            <> if isActiveParticipant then
+                [ "bg-gradient-to-r", "from-blue", "to-lightblue", "text-white" ]
+              else
+                [ "bg-gray", "text-black", "opacity-50", "cursor-default" ]
+        , enabled isActiveParticipant
+        -- FIXME
+        -- , onClick_ $ NEEDACTION
+        ]
+        [ span_ [ text "Deposit:" ]
+        -- FIXME: Install purescript-formatters to separate by thousands
+        , span_ [ currency token value ]
+        ]
+    ]
+
+renderAction isActiveParticipant (MakeChoice choiceId bounds chosen) =
+  let
+    bigNumberToInt = floor <<< toNumber
+
+    options :: Array Int
+    options =
+      nub
+        $ bounds
+        >>= \(Bound fromB toB) -> range (bigNumberToInt fromB) (bigNumberToInt toB)
```

Comment:
I imagine this function will go away when changing to inputs, but also `bigNumberToInt` defeats the purpose of using `bigNumber`. It could work for the prototype but we would need a FIXME at least, so that we remember to change it

Reply:
I was considering to be an enum so I thought big number was an overkill... Now that I'll set them as input they wont be needed. Will remove in the next PR